src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -531,6 +531,14 @@ void SweepDebugInfo (Collection<MethodDefinition> methods)
 
 		protected virtual bool ShouldRemove<T> (T element) where T : IMetadataTokenProvider
 		{
+			if (typeof (T) == typeof (ExportedType)) {
+				var typeDef = (element as ExportedType).Resolve ();
```

Comment:
If you need ExportedType special it'd be nicer to introduce SweepCollectionExportedTypes method